src/libcore/fmt/mod.rs

```diff
@@ -844,12 +844,36 @@ impl Display for char {
     }
 }
 
+// Two extra bytes for 0x
+#[cfg(target_pointer_width = "32")]
+const POINTER_PADDING: Option<usize> = Some(10);
+#[cfg(target_pointer_width = "64")]
+const POINTER_PADDING: Option<usize> = Some(18);
```

Comment:
I'm more worried that `#[cfg]` code isn't typechecked at all if it's for a different platform, often leading to really annoying cross-platform bugs.